scripts/python/migrate_data_2024.py

```diff
@@ -0,0 +1,2683 @@
+"""
+def dump_users(host, port, db, user, password, attribs):
+    users_data = {}
+
+    sql_query_user = f""" SELECT username, email, panel_attrib
+                          FROM user """
+    
+    connection = mysql.connector.connect(host=host,
+                                         database=db,
+                                         user=user,
+                                         port=port,
+                                         password=password)
+
+    try:
+        if connection.is_connected():
+            cursor = connection.cursor()
+            cursor.execute(sql_query_user)
+            data = cursor.fetchall()
+            if len(data) != 0:
+                for row in data:
+                    # print(f"user: {row[0]}, email: {row[1]}, panel_attrib: {list(row[2])}")
+                    panel_names = []
+                    for panel in list(row[2]):
+                        # print(f"attrib: {panel}, {attribs[int(panel)]['attrib_value']}")
+                        panel_names.append(attribs[int(panel)]['attrib_value'])
+                    users_data[row[0]] = { 'email':row[1],
+                                           'panels':panel_names }
+
+    except Error as e:
+        print("Error while connecting to MySQL", e)
+    finally:
+        if connection.is_connected():
+            cursor.close()
+            connection.close()
+
+    return users_data
+
+"""
+    Return the publications
+"""
+def dump_publications(host, port, db, user, password):
+    publications_data = {}
+    duplicated_pmids = {}
+
+    sql_query = f""" SELECT publication_id, pmid, title, source
+                     FROM publication """
+    
+    sql_query_duplicates = f""" SELECT pmid 
+                                FROM publication 
+                                WHERE pmid IS NOT NULL
+                                GROUP BY pmid HAVING count(*)>1
+                            """
+
+    sql_query_gfd = f""" SELECT distinct publication_id
+                         FROM genomic_feature_disease_publication
+                         WHERE publication_id in ( SELECT publication_id FROM publication WHERE pmid = %s ) """
+
+    connection = mysql.connector.connect(host=host,
+                                         database=db,
+                                         user=user,
+                                         port=port,
+                                         password=password)
+
+    try:
+        if connection.is_connected():
+            cursor = connection.cursor()
+            cursor.execute(sql_query_duplicates)
+            data_dup = cursor.fetchall()
+            if len(data_dup) != 0:
+                for row in data_dup:
+                    pmid = row[0]
+                    cursor.execute(sql_query_gfd, [pmid])
+                    data_gfd = cursor.fetchall()
+                    if len(data_gfd) == 0:
+                        duplicated_pmids[pmid] = 1
```

Comment:
if len(data_gfd) == 0:
                        duplicated_pmids[pmid] = 1
Comment
That would also mean this would be changed to 
 if len(data_gfd) == 0:
                        duplicated_pmids[row[0]] = 1
Only suggesting because the assigned variable is not used anywhere else except in the for block